### PR TITLE
[docs] Add information about retrieving balance changes from events

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
This describes how to retrieve balance changes from transactions (Similar to the Rosetta implementation).
 
 Additionally, updates with the new event GUIDs

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4449)

### CODE REVIEW

**Review by rajkaramchedu — APPROVED** [review]

**Review by hardsetting — APPROVED** [review]
Looks good

File: developer-docs-site/docs/guides/system-integrators-guide.md [hunk 1]
```diff
@@ -386,6 +412,129 @@ When tracking full movement of coins, normally events are sufficient. `0x1::apto
...
+
+:::tip
+If you always use the events in this manner, you won't miss any balance changes for an account.
+By monitoring the events, it will include any balance changes in the `0x1::coin::CoinStore`:
+1. Coin mints
+2. Coin burns
+3. Coin transfers
+4. Staking coins
+5. Withdrawing staked coins
+6. Non-standard transfers
```
Comment (davidiw) [inline developer-docs-site/docs/guides/system-integrators-guide.md:534]: maybe say e.g., those that do not derive from `coin::transfer`

File: developer-docs-site/docs/guides/system-integrators-guide.md [hunk 2]
```diff
@@ -358,6 +373,10 @@ Each event in `events` is differentiated by an `key`. The `key` is derived from
 
 Using the above example, `events[1].key` is `0x02000000000000005098df8e7969b58ab3bd2d440c6203f64c60a1fd5c08b9d4abe6ae4216246c3e`. This is equivalent to `changes[0].data.data.deposit_events.guid`, which is `{"addr": "0x5098df8e7969b58ab3bd2d440c6203f64c60a1fd5c08b9d4abe6ae4216246c3e", "creation_num": "2"}`.
 
+:::tip
```
Comment (davidiw) [inline developer-docs-site/docs/guides/system-integrators-guide.md:376]: line 357 to 359 above (Each event in `events`...) needs to be cleaned up. Perhaps we can just get rid of this nonsense and say how guid => guid?

**Review by davidiw — COMMENTED** [review]
looks good and really appreciate the extra example :)

**Pull request merged**